include/zephyr/sys/util.h

```diff
@@ -210,6 +210,26 @@ extern "C" {
 #define CONTAINER_OF(ptr, type, field) \
 	((type *)(((char *)(ptr)) - offsetof(type, field)))
 
+/**
+ * @brief Get the size of a field in a structure type
+ *
+ * Example:
+ *
+ *	struct foo {
+ *		int bar;
+ *		char baz;
+ *	};
+ *
+ *	size_t size_of_baz = SIZEOF_FIELD(struct foo, baz);
+ *
+ * Above, @p size_of_baz is equal to sizeof(char).
+ *
+ * @param type the type of the structure containing the field
+ * @param member the name of the field in the structure
+ * @return the size of the field in bytes
+ */
+#define SIZEOF_FIELD(type, member) sizeof(((type *)0)->member)
```

Comment:
I have no strong opinion on this naming choice, but looking at the other macro names, shouldn't it be called `FIELD_SIZE` (cf. `ARRAY_SIZE`)?

Reply:
Ah, sorry, I just saw that this names comes from @M1cha's advice with reference to the Linux kernel.